client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -1122,6 +1123,26 @@ static Request analyze(AnalyzeRequest request) throws IOException {
         return req;
     }
 
+    static Request termVectors(TermVectorsRequest tvrequest) throws IOException {
+        String endpoint = "_termvectors";
+        if (tvrequest.getId() != null) {
+            endpoint = new EndpointBuilder().addPathPart(
+                tvrequest.getIndex(), tvrequest.getType(), tvrequest.getId()).addPathPartAsIs(endpoint).build();
+        } else {
+            endpoint = new EndpointBuilder().addPathPart(tvrequest.getIndex(), tvrequest.getType()).addPathPartAsIs(endpoint).build();
+        }
+        Request request = new Request(HttpGet.METHOD_NAME, endpoint);
+        Params params = new Params(request);
+        if (tvrequest.getRouting() != null) params.withRouting(tvrequest.getRouting());
```

Comment:
the params methods all call `putParam` and that does a null check, and will not add it if its null. So no need to check here.